src/SFA.DAS.ApplyService.Web/ViewModels/Roatp/TaskListViewModel.cs

```diff
@@ -48,11 +52,183 @@ public string SectionStatus(int sequenceId, int sectionId)
             return RoatpTaskListWorkflowService.SectionStatus(ApplicationSequences, NotRequiredOverrides, sequenceId, sectionId, ApplicationRouteId);
         }
 
+        public string WhosInControlCss
+        {
+            get
+            {
+                var status = WhosInControlSectionStatus;
+
+                if (status == String.Empty)
+                {
+                    return "hidden";
+                }
+
+                var cssClass = status.ToLower();
+                cssClass = cssClass.Replace(" ", "");
+
+                return cssClass;
+            }
+        }
+
+        public string WhosInControlSectionStatus
+        {
+            get
+            {
+                if (VerifiedCompaniesHouse && VerifiedCharityCommission)
+                {
+                    if ((CompaniesHouseDataConfirmed && !CharityCommissionDataConfirmed)
+                        || (!CompaniesHouseDataConfirmed && CharityCommissionDataConfirmed))
+                    {
+                        return "In Progress";
+                    }
+                    if (CompaniesHouseDataConfirmed && CharityCommissionDataConfirmed)
```

Comment:
I'm not sure all of these if statements are efficient. Couldn't you use if/else if or move to a switch ?

Are there unit tests to cover all of this logic ?

Reply:
There are unit tests for this. The code has to cover cases where a provider can be a company, a charity or and company and a charity.